src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -22,14 +22,14 @@ import scala.collection.generic.SerializeEnd
 import scala.collection.mutable.{ArrayBuffer, Builder, ReusableBuilder, StringBuilder}
 import scala.language.implicitConversions
 
-/**  The class `LazyList` implements lazy lists where elements
-  *  are only evaluated when they are needed. Here is an example:
+/**  This class implements an immutable linked list that evaluates elements
+  *  in order and only when needed. Here is an example:
```

Comment:
this should really explain that it's a lazy state and how that works

Reply:
I would also make it more explicit what “when needed” mean, by mentioning at least `isEmpty`, `head`, `tail` and reduction operations such as `foldLeft` or `size`.

Reply:
@julienrf do you think it would be a good idea to talk about it in terms of being either empty or a cons cell, or remain more abstract?

Reply:
I would just enumerate the most common operations that do evaluate elements.

Reply:
thoughts on this wording so far?

```scala
/**  This class implements an immutable linked list that evaluates its states in order
  *  and only when needed. A `LazyList` can have one of two states:
  *     - empty
  *     - a [[head `head`]] element and [[tail `tail`]] `LazyList`
  *  If a `LazyList` is not empty, its `tail` is another `LazyList` whose state is
  *  evaluated lazily, and if the `tail` is non-empty, its `tail` is another `LazyList`
  *  whose state is evaluated lazily, and so on.
  *  Whether a `LazyList` is empty or not is only evaluated when that information is needed
  *  for a particular operation; for example, `head`, `tail` and [[isEmpty `isEmpty`]] all
  *  evaluate the state of of the `LazyList` (but not the states of any `tail`s if the list
  *  is non-empty).
  *
  *
  *  Here is an example:
  *
  *  {{{
  *  // ...
```

Reply:
The key difference between `LazyList` and `Stream` - and its key feature - is that whether or not it is lazy is evaluated lazily. I'm not sure how best to convey that.

Reply:
ping @julienrf @szeiger